neard/src/cli.rs

```diff
@@ -817,6 +818,62 @@ impl ValidateConfigCommand {
     }
 }
 
+#[cfg(target_os = "linux")]
+fn normalize_whitespace(s: &str) -> String {
+    s.split_whitespace().collect::<Vec<_>>().join(" ")
+}
+
+/// Checks the provided kernel parameter  from /proc/sys
+/// and prints an error if it is not set to the expected value.
+#[cfg(target_os = "linux")]
+fn check_kernel_param(param_name: &str, expected_value: &str) {
+    // Convert the dotted param_name into a path under /proc/sys
+    // For example, "net.core.rmem_max" -> "/proc/sys/net/core/rmem_max"
+    let mut path = PathBuf::from("/proc/sys");
+    for part in param_name.split('.') {
+        path.push(part);
+    }
+
+    match std::fs::read_to_string(&path) {
+        Ok(contents) => {
+            let actual = contents.trim();
+            let actual_normalized = normalize_whitespace(actual);
+            let expected_normalized = normalize_whitespace(expected_value);
+
+            if actual_normalized != expected_normalized {
+                error!(
+                    "ERROR: {} is set to {}, expected {}",
+                    param_name, actual_normalized, expected_normalized
```

Comment:
Should we point the users to run scripts/set_kernel_params.sh in case of mismatch?

Reply:
yup, thanks